docs/resources/pumls/model.puml

```diff
@@ -0,0 +1,58 @@
+@startuml Model
+
+!includeurl ./style.puml
+
+package "Model" {
+
+    class ModelMetrics {
+        - _accuracy: float
+        - _precision: float
+        - _recall: float
+        - _f1_score: float
+        - _loss: float
+        + accuracy(): float
+        + precision(): float
+        + recall(): float
+        + f1_score(): float
+        + loss(): float
+    }
+
+    class ModelDetails {
+        - _name: str
+        - _version: str
+        - _framework: str
+        - _target: str
+        - _dtype: str
+        + name(): str
+        + version(): str
+        + framework(): str
+        + target(): str
+        + dtype(): str
+    }
+
+    class ModelInfo {
+        - _id: uuid.UUID
+        - _details: ModelDetails
+        - _uploaded_date: datetime
+        - _metrics: ModelMetrics
+        + id(): uuid.UUID
+        + id(new_id: uuid.UUID): void
+        + details(): ModelDetails
+        + uploaded_date(): datetime
+        + metrics(): ModelMetrics
+    }
+
+    class Model {
```

Comment:
Add `__init__(model_info: ModelInfo, model_binary: bytes)` and `__iter__():Iterator[Tuple[str, Any]]` functions